### ISSUE TITLE
Why basic auth can't work

### ISSUE THREAD

**Issue by thlo7777** [RESTful-Drupal/restful#928]
I'm a newbie with restful. 

I modify sites/all/modules/restful_example/src/Plugin/resource/node/article/v2/Articles__2_0.php

/**
- Class Articles
- @package Drupal\restful\Plugin\resource
  *
- @Resource(
-   name = "articles:2.0",
-   resource = "articles",
-   label = "Articles",
-   description = "Export the article content type.",
-   authenticationTypes = {
-     "basic_auth"
-   },
-   authenticationOptional = FALSE,
-   dataProvider = {
-     "entityType": "node",
-     "bundles": {
-       "article"
-     },
-   },
-   majorVersion = 2,
-   minorVersion = 0
- )
  */

Only changed authenticationTypes and authenticationOptional followed by  [restful guide](https://www.youtube.com/watch?v=5XryKGAXuLw&list=PLZOQ_ZMpYrZv8_c7jd_CkO_93-DnyVFY5&index=20) guide.

Then I use http://dld.dreamland360.com/api/v2.0/articles to access server.  the result still give me all articles instead of 403 or 400 error.  
Has anyone can give me clues 

Thanks a lot

**Comment by e0ipso** [comment]
Did you send the `Authorization: Basic XXXX` header with your request to http://dld.dreamland360.com/api/v2.0/articles?

**Comment by thlo7777** [comment]
Thanks a lot e0ipso 👍 . 
Actually I used chrome browser and put http://dld.dreamland360.com/api/v2.0/articles to address bar. I guess it would not send Authorization: Basic XXXX in header ? 

By the way I follow document and restful_test module, create a my own module to implement basic auth feature, it's ok for me.  The new module return {"type":"http:\/\/www.w3.org\/Protocols\/rfc2616\/rfc2616-sec10.html#sec10.4.2","title":"Bad credentials. Anonymous user resolved for a resource that requires authentication.","status":401,"detail":"Unauthorized","instance":"http:\/\/dld.dreamland360.com\/help\/restful\/problem-instances-unauthorized"}

But I can't understand just restful_example can't work for basic auth. 
My environment : Drupal 7.34 + restful 7.x-2.11   and turn on restful_example and restful_tests module

code is 
- Contains \Drupal\restful_example\Plugin\resource\node\article\v2\Articles__2_0.
  */

namespace Drupal\restful_example\Plugin\resource\node\article\v2;

use Drupal\restful\Plugin\resource\ResourceInterface;
use Drupal\restful\Plugin\resource\ResourceNode;

/**
- Class Articles
- @package Drupal\restful\Plugin\resource
  *
- @Resource(
-   name = "articles:2.0",
-   resource = "articles",
-   label = "Articles",
-   description = "Export the article content type.",
-   authenticationTypes = {
-     "basic_auth"
-   },
-   authenticationOptional = FALSE,
-   dataProvider = {
-     "entityType": "node",
-     "bundles": {
-       "article"
-     },
-   },
-   majorVersion = 2,
-   minorVersion = 0
- )
  */
  class Articles__2_0 extends ResourceNode implements ResourceInterface {
  $public_fields = parent::publicFields();
  
    $public_fields['body'] = array(
      'property' => 'body',
      'sub_property' => 'value',
    );
  }

Chrome address bar url is "http://dld.dreamland360.com/api/v2.0/articles"

and reture value is

{"data":[{"type":"articles","id":"5","attributes":{"id":5,"label":"restful test","self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/5"},"links":{"self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/5"}},{"type":"articles","id":"14","attributes":{"id":14,"label":"\u6ce8\u518c\u8868\u63d0\u9192","self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/14"},"links":{"self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/14"}},{"type":"articles","id":"29","attributes":{"id":29,"label":"tutututututut","self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/29"},"links":{"self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles\/29"}}],"meta":{"count":3},"links":{"self":"http:\/\/dld.dreamland360.com\/api\/v2.0\/articles?page[number]=1"}}

Thank you for support

**Comment by thlo7777** [comment]
Sorry e0ipso
I found root cause. It's my fault. I copy restful_example sub folder from restful module to sites/all/modules/restful_example. Actually I have two restful_example modules in drupal.  One located in restful module, the other located in sites/all/modules. I only modified the other restful_example, but drupal always call it which is in restful folder. That was my misunderstanding. 

After I deleted sites/all/modules/restful_example folder. Everything is ok.  

Thank a lot for take your time to help me.
